datafusion/sql/src/unparser/dialect.rs

```diff
@@ -157,6 +157,15 @@ pub trait Dialect: Send + Sync {
     fn full_qualified_col(&self) -> bool {
         false
     }
+
+    /// Allow to unparse the unnest plan as [ast::TableFactor::UNNEST].
+    ///
+    /// Some dialects like BigQuery require UNNEST to be used in the FROM clause but
+    /// the LogicalPlan planner always put UNNEST in the SELECT clause. This flag allows
```

Comment:
put -> puts

well, UNNEST is a lateral relational operator. It doesn't belong to the select clause.
It's probably yet another witness of the problem that the LP is half way between AST and a good IR (which we don't have yet)

Reply:
> well, UNNEST is a lateral relational operator. It doesn't belong to the select clause. It's probably yet another witness of the problem that the LP is half way between AST and a good IR (which we don't have yet)
>

Interesting.
The logical planner plans `FROM unnest([1,2,3])` to
```
Projection: unnest_placeholder(make_array(Int64(1),Int64(2),Int64(3)),depth=1) AS UNNEST(make_array(Int64(1),Int64(2),Int64(3)))
    Unnest: lists[unnest_placeholder(make_array(Int64(1),Int64(2),Int64(3)))|depth=1] structs[]
      Projection: make_array(Int64(1), Int64(2), Int64(3)) AS unnest_placeholder(make_array(Int64(1),Int64(2),Int64(3)))
        EmptyRelation
``` 
Do you think it's a good plan? What would a good IR look like? 🤔

Reply:
It's not unreasonable.
Looking at this the "unnest_placeholder" looks more like "unnest input".